### PR TITLE
feat: add PeekKeys and PeekTrailerKeys

### PR DESCRIPTION
**Pull request by li-jin-gou**
pr for https://github.com/valyala/fasthttp/issues/1400

### CODE REVIEW

File: header.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ type ResponseHeader struct {
 	contentEncoding []byte
 	server          []byte
 	mulHeader       [][]byte
+	keys            [][]byte
```
Comment (erikdubbelboer) [inline header.go:46]: You can re-use `mulHeader` for this. But I do think we need to improve the warning for these and the `PeekAll` functions to make it clear that any further calls to `Peek*` functions will also change the return values.

**Review by erikdubbelboer — CHANGES_REQUESTED** [review]

File: header.go (hunk 1 above)
Comment (li-jin-gou) [reply header.go]: I have modified the code and added the relevant code comments

File: header.go [hunk 2]
```diff
@@ -1801,6 +1801,7 @@ func (h *RequestHeader) peek(key []byte) []byte {
 //
 // The returned value is valid until the request is released,
 // either though ReleaseRequest or your request handler returning.
+// Any call to the Peek* function will change the return values.
```
Comment (erikdubbelboer) [inline header.go:1804]:
```suggestion
// Any future calls to the Peek* will modify the returned value.
```

File: header.go [hunk 3]
```diff
@@ -1847,7 +1848,8 @@ func (h *RequestHeader) peekAll(key []byte) [][]byte {
 // PeekAll returns all header value for the given key.
 //
 // The returned value is valid until the request is released,
-// either though ReleaseRequest or your request handler returning.
+// either though ReleaseResponse or your request handler returning.
+// Any call to the Peek* function will change the return values.
```
Comment (erikdubbelboer) [inline header.go:1852]:
```suggestion
// Any future calls to the Peek* will modify the returned value.
```

File: header.go [hunk 4]
```diff
@@ -1887,6 +1889,54 @@ func (h *ResponseHeader) peekAll(key []byte) [][]byte {
 	return h.mulHeader
 }
 
+// PeekKeys return all header keys.
+//
+// The returned value is valid until the request is released,
+// either though ReleaseRequest or your request handler returning.
+// Any call to the Peek* function will change the return values.
```
Comment (erikdubbelboer) [inline header.go:1896]:
```suggestion
// Any future calls to the Peek* will modify the returned value.
```

File: header.go [hunk 5]
```diff
@@ -1887,6 +1889,54 @@ func (h *ResponseHeader) peekAll(key []byte) [][]byte {
...
+	h.mulHeader = h.mulHeader[:0]
+	h.mulHeader = peekArgsKeys(h.mulHeader, h.h)
+	return h.mulHeader
+}
+
+// PeekTrailerKeys return all trailer keys.
+//
+// The returned value is valid until the request is released,
+// either though ReleaseRequest or your request handler returning.
+// Any call to the Peek* function will change the return value.
```
Comment (erikdubbelboer) [inline header.go:1908]:
```suggestion
// Any future calls to the Peek* will modify the returned value.
```

File: header.go [hunk 6]
```diff
@@ -1887,6 +1889,54 @@ func (h *ResponseHeader) peekAll(key []byte) [][]byte {
...
+	h.mulHeader = h.mulHeader[:0]
+	h.mulHeader = peekArgsKeys(h.mulHeader, h.trailer)
+	return h.mulHeader
+}
+
+// PeekKeys return all header keys.
+//
+// The returned value is valid until the request is released,
+// either though ReleaseResponse or your request handler returning.
+// Any call to the Peek* function will change the return values.
```
Comment (erikdubbelboer) [inline header.go:1920]:
```suggestion
// Any future calls to the Peek* will modify the returned value.
```

File: header.go [hunk 7]
```diff
@@ -1887,6 +1889,54 @@ func (h *ResponseHeader) peekAll(key []byte) [][]byte {
...
+	h.mulHeader = h.mulHeader[:0]
+	h.mulHeader = peekArgsKeys(h.mulHeader, h.h)
+	return h.mulHeader
+}
+
+// PeekTrailerKeys return all trailer keys.
+//
+// The returned value is valid until the request is released,
+// either though ReleaseResponse or your request handler returning.
+// Any call to the Peek* function will change the return values.
```
Comment (erikdubbelboer) [inline header.go:1932]:
```suggestion
// Any future calls to the Peek* will modify the returned value.
```

**Review by erikdubbelboer — CHANGES_REQUESTED** [review]

**Review by erikdubbelboer — APPROVED** [review]

File: header.go (hunk 2 above)
Comment (li-jin-gou) [reply header.go]: thanks

**Comment by erikdubbelboer** [comment]
Thanks!

**Pull request merged**